File: src/main/java/com/wafflestudio/draft/dto/response/RoomsOfUserResponse.java
package com.wafflestudio.draft.dto.response;

import com.wafflestudio.draft.model.Room;
import com.wafflestudio.draft.model.User;
import lombok.Data;

import java.util.ArrayList;
import java.util.List;

@Data
public class RoomsOfUserResponse {
    private Long id;
    private String username;
    private String email;
    private List<RoomResponse> rooms;

    public RoomsOfUserResponse(User user) {
        this.id = user.getId();
        this.username = user.getUsername();
        this.email = user.getEmail();
    }
}

File: src/main/kotlin/com/wafflestudio/draft/dto/response/RoomsOfUserResponse.kt
package com.wafflestudio.draft.dto.response

import com.wafflestudio.draft.dto.RoomDTO
import com.wafflestudio.draft.model.User

data class RoomsOfUserResponse(
        var id: Long? = null,
        var username: String? = null,
        var email: String? = null,
        var rooms: List<RoomDTO.Response>? = null
) {
    constructor(user: User) : this(
            user.id,
            user.username,
            user.email
    )
}
